### ISSUE TITLE
Add `SaveEnsembleMixin`

### ISSUE THREAD

**Issue by Mr-Geekman** [tinkoff-ai/etna#1043]
### 🚀 Feature Request

We want to make ensembles saveable.

### Proposal

1. Add class `SaveEnsembleMixin(SaveMixin)`, it will be responsible for saving all ensembles.
2. Signature of save for this moment remains the same. But in the future we plan to make it different.
    ```python
    def save(self, path: pathlib.Path, save_ts: bool = False).
    ```
    It will make it possible to save `ts` with pipeline. For now, we don't want to implement saving `TSDataset` (it will be much simplier after completion of TSDataset-v2.0).
3. Signature of load we want to change.
    ```python
    def load(cls, path: pathlib.Path, ts: Optional[TSDataset] = None) -> Any.
    ```
    We expect that user gives us dataset we want to work with. If user hasn't give us the dataset, may be he wants to refit pipeline or he want to give a custom `TSDataset` into `forecast`/`predict`.
4. How save should work.
    * Remove attributes `pipelines`, `ts`.
    * Save object with `SaveMixin`.
    * Save each pipeline in `pipelines` separately into folder inside the archive.
5. How load should work.
    * Load object with `SaveMixin`.
    * Load `pipelines`. We should get classname from meta or implement `load` method that makes this logic with loading classname for us.

### Test cases

1. Test mixin.
    * Test in saving that everything except for pipeline is saved in `object.pkl`.
    * Test in saving that pipelines are saved into separate folder and can be loaded from it.
    * Test error on non-existent file.
    * Test that load restores pipelines and ts (if given).
    * Test warning on different versions on etna.
2. Test saving/loading on `VotingEnsemble`.
3. Test saving/loading on `StackingEnsemble`.
4. Test saving/loading on `DirectEnsemble`.

**Comment by Mr-Geekman** [comment]
Closed wia #1046.

**Comment by Mr-Geekman** [comment]
Closed via https://github.com/tinkoff-ai/etna/pull/1046.
